boilerplate/app/models/helpers/use-stores.ts

```diff
@@ -0,0 +1,80 @@
+import { createContext, useContext, useEffect, useState } from "react"
+import { setReactotronRootStore } from "../../services/reactotron"
+import { RootStore, RootStoreModel } from "../RootStore"
+import { setupRootStore } from "./setup-root-store"
+
+/**
+ * Create the initial (empty) global RootStore instance here.
+ *
+ * Later, it will be rehydrated in app.tsx with the setupRootStore function.
+ *
+ * If your RootStore requires specific properties to be instantiated,
+ * you can do so here.
+ *
+ * If your RootStore has a _ton_ of sub-stores and properties (the tree is
+ * very large), you may want to use a different strategy than immediately
+ * instantiating it, although that should be rare.
+ */
+const _rootStore = RootStoreModel.create({})
+
+/**
+ * The RootStoreContext provides a way to access
+ * the RootStore in any screen or component.
+ */
+const RootStoreContext = createContext<RootStore>(_rootStore)
+
+/**
+ * You can use this Provider to specify a *different* RootStore
+ * than the singleton version above if you need to. Generally speaking,
+ * this Provider & custom RootStore instances would only be used in
+ * testing scenarios.
+ */
+export const RootStoreProvider = RootStoreContext.Provider
+
+/**
+ * A hook that screens and other components can use to gain access to
+ * our stores:
+ *
+ * const rootStore = useStores()
+ *
+ * or:
+ *
+ * const { someStore, someOtherStore } = useStores()
+ */
+export const useStores = () => useContext(RootStoreContext)
+
+/**
+ * Used only in the app.tsx file, this hook sets up the RootStore
+ * and then rehydrates it. It connects everything with Reactotron
+ * and then lets the app know that everything is ready to go.
+ */
+export const useInitialRootStore = (callback: () => void) => {
```

Comment:
callback should probably be typed as `() => Promise<void>` just in case.

Reply:
Good call